concepts/type-conversion/about.md

```diff
@@ -1,10 +1,272 @@
+The result is what you would expect it to be for primitive values.
+
+```javascript
+String(12.34);
+// => '12.34'
+
+String(false);
+// => 'false'
+
+String(null);
+// => 'null'
+
+String(undefined);
+// => 'undefined'
+```
+
+For arrays, the `String` function will apply the string conversion for each element and join the results with a comma.
+You can also apply the [`join` method][mdn-join] yourself, e.g. to customize the separator.
+However, in these cases `null` and `undefined` will be converted to an empty string.
+
+```javascript
+String([42, null, true, 'abc']);
+// => '42,,true,abc'
+```
+
+For objects, by default `String` returns an unhelpful text.
+
+```javascript
+String({ key: 'value' });
+// => '[object Object]'
+```
+
+You can customize the conversion behavior, e.g. by providing a `toString` method.
+The section "[Object to primitive conversion][custom-conversion]" on javascript.info explains the details.
+
+Another common way to achieve a better string representation for objects and arrays is to use [JSON encoding][json].
+
+```javascript
+const obj = {
+  name: 'Gilda Guerra',
+  address: {
+    city: 'Goiânia',
+  },
+};
+
+JSON.stringify(obj);
+// => '{"name":"Gilda Guerra","address":{"city":"Goiânia"}}'
+```
+
+## Type Coercion
+
+In certain contexts, JavaScript will automatically convert a value to another data type before it evaluates some statement.
+This implicit conversion is called _type coercion_.
+
+### Boolean Context
+
+When a value is used in a boolean context, JavaScript will apply the same rules as the `Boolean` function to implicitly convert the value.
+
+- When the condition of an [if statement][concept-conditionals] is not a boolean, coercion is applied to determine whether the condition is fulfilled or not.
+  The same applies for the first operand of the [ternary operator][mdn-ternary] `?`.
+
+  ```javascript
+  const num = 0;
+  if (num) {
+    // this block NOT is executed because 0 is falsy
+  }
+
+  const name = 'Jin';
+  name ? 'name was provided' : 'no name provided';
+  // => 'name was provided'
+  ```
+
+- The operand of the logical NOT operator `!` is also coerced into boolean before the NOT operation is applied.
+
+  ```javascript
+  const name = '';
+  !name;
+  // => true
+  ```
+
+  A result of the described behavior is that `!!value` has the same effect as `Boolean(value)`.
+  Nevertheless, you should use `Boolean` for readability.
+
+- JavaScript also applies coercion for the operands of the logical AND (`&&`) and OR (`||`) operators.
+  But keep in mind that the result of the expression is **not** necessarily a boolean.
+  It returns one of the original operands (see [MDN on Logical Operators][mdn-logical-operators]).
+
+  ```javascript
+  null || 'hello';
+  // => 'hello'
+  ```
+
+### String Context
+
+If the addition operator `+` is used for primitive values and one operand is a string, the other one will be coerced into a string as well (if necessary).
```

Comment:
I think the "if necessary" is already implied by saying "one operand is a string". If only one of the is a string, it is necessary.

```suggestion
If the addition operator `+` is used for primitive values and one operand is a string, the other one will be coerced into a string as well.
```

Reply:
fixed in all the places